File: Python/calculadora_imc.py
#leer datos de entrada
print('Ingrese su peso en kg: ')
peso = float(input())
print('Ingrese su altura en metros: ')
altura = float(input()) 

#validad el peso <= 400 y >=5
if peso>=5 and peso<=400:
    #validar la altura <= 2.5 y >= 0.5
    if altura>=0.5 and altura<=2.5:
        #calcular el imc
        imc = peso / (altura ** 2)
        print('Tu IMC es: ', imc)

        #clasificar el imc
        if imc < 18.5:
            print('Bajo peso')
        elif imc >= 18.5 and imc < 25:
            print('Peso normal')
        elif imc >= 25 and imc < 30:
            print('Sobrepeso')
        elif imc >= 30:
            print('Obesidad')
    else:
        print('Altura Ivalidad')
else:
    print('Peso Ivalidado')

File: C#/calculadora_imc.cs
Console.WriteLine("Ingrese su peso en kg: ");
float peso = float.Parse(Console.ReadLine());
Console.WriteLine("Ingrese su altura en metros: ");
float altura = float.Parse(Console.ReadLine());

if (peso >= 5 && peso <= 400)
{
    if (altura >= 0.5 && altura <= 2.5)
    {
    float imc = peso / (altura * altura);
    Console.WriteLine("Su IMC es: " + imc);
    if (imc < 18.5)
    {
        Console.WriteLine("Bajo peso");
    }
    else if (imc >= 18.5 && imc < 25)
    {
        Console.WriteLine("Peso normal");
    }
    else if (imc >= 25 && imc < 30)
    {
        Console.WriteLine("Sobrepeso");
    }
    else
    {
        Console.WriteLine("Obesidad");
    }
    }
    else
    {
        Console.WriteLine("Altura Invalida");
    }
}
else
{
    Console.WriteLine("Altura Invalidao");
}